From 83c106b8e91ceb603bb1f88852c7481ac5c7c356 Mon Sep 17 00:00:00 2001
From: Javier Pena <jpena@redhat.com>
Date: Wed, 11 Feb 2015 11:08:07 +0100
Subject: [PATCH] Fix prefetch refresh for providers

The instance method for most providers did only gather instances
during the first execution. After that, usage of ||= prevented it
from gathering them again, which broke cases like creating a tenant
and then trying to fetch that tenant's id for other usage.

Change-Id: I5a9767383438549e354429632f8c1a7e569aa8fb
---
 keystone/lib/puppet/provider/keystone_endpoint/openstack.rb | 2 +-
 keystone/lib/puppet/provider/keystone_role/openstack.rb     | 2 +-
 keystone/lib/puppet/provider/keystone_service/openstack.rb  | 2 +-
 keystone/lib/puppet/provider/keystone_tenant/openstack.rb   | 2 +-
 keystone/lib/puppet/provider/keystone_user/openstack.rb     | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/keystone/lib/puppet/provider/keystone_endpoint/openstack.rb b/keystone/lib/puppet/provider/keystone_endpoint/openstack.rb
index fadb25f..434742e 100644
--- a/keystone/lib/puppet/provider/keystone_endpoint/openstack.rb
+++ b/keystone/lib/puppet/provider/keystone_endpoint/openstack.rb
@@ -108,7 +108,7 @@ Puppet::Type.type(:keystone_endpoint).provide(
   end
 
   def instance(name)
-    @instances ||= instances.select { |instance| instance[:name] == name }.first || {}
+    @instances = instances.select { |instance| instance[:name] == name }.first || {}
   end
 
   def flush
diff --git a/keystone/lib/puppet/provider/keystone_role/openstack.rb b/keystone/lib/puppet/provider/keystone_role/openstack.rb
index 50a2cbb..3f44fd3 100644
--- a/keystone/lib/puppet/provider/keystone_role/openstack.rb
+++ b/keystone/lib/puppet/provider/keystone_role/openstack.rb
@@ -46,7 +46,7 @@ Puppet::Type.type(:keystone_role).provide(
   end
 
   def instance(name)
-    @instances ||= instances.select { |instance| instance[:name] == name }.first || {}
+    @instances = instances.select { |instance| instance[:name] == name }.first || {}
   end
 
 end
diff --git a/keystone/lib/puppet/provider/keystone_service/openstack.rb b/keystone/lib/puppet/provider/keystone_service/openstack.rb
index e9bed96..b4be984 100644
--- a/keystone/lib/puppet/provider/keystone_service/openstack.rb
+++ b/keystone/lib/puppet/provider/keystone_service/openstack.rb
@@ -82,7 +82,7 @@ Puppet::Type.type(:keystone_service).provide(
   end
 
   def instance(name)
-    @instances ||= instances.select { |instance| instance[:name] == name }.first || {}
+    @instances = instances.select { |instance| instance[:name] == name }.first || {}
   end
 
   def flush
diff --git a/keystone/lib/puppet/provider/keystone_tenant/openstack.rb b/keystone/lib/puppet/provider/keystone_tenant/openstack.rb
index b0a6361..0ee3b55 100644
--- a/keystone/lib/puppet/provider/keystone_tenant/openstack.rb
+++ b/keystone/lib/puppet/provider/keystone_tenant/openstack.rb
@@ -83,7 +83,7 @@ Puppet::Type.type(:keystone_tenant).provide(
   end
 
   def instance(name)
-    @instances ||= instances.select { |instance| instance[:name] == name }.first || {}
+    @instances = instances.select { |instance| instance[:name] == name }.first || {}
   end
 
   def flush
diff --git a/keystone/lib/puppet/provider/keystone_user/openstack.rb b/keystone/lib/puppet/provider/keystone_user/openstack.rb
index f91b3b0..3a87f04 100644
--- a/keystone/lib/puppet/provider/keystone_user/openstack.rb
+++ b/keystone/lib/puppet/provider/keystone_user/openstack.rb
@@ -195,7 +195,7 @@ Puppet::Type.type(:keystone_user).provide(
   end
 
   def instance(name)
-    @instances ||= instances.select { |instance| instance[:name] == name }.first || {}
+    @instances = instances.select { |instance| instance[:name] == name }.first || {}
   end
 
   def set_project(newproject)
